### ISSUE TITLE
New mission objective: stay in system for X seconds

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#4475]
**Is your feature request related to a problem? Please describe.**
The timer ship in the Sad Archie mission is an interesting use of game mechanics, but in my attempts to generalize its use to simply require the player to be in a system for a set amount of time, it's proven rather inconsistent, either because the timer ship has difficulty damaging itself, or because it is possible to use the blast radius of an explosion to kill a timer ship.

**Describe the solution you'd like**
In order to prevent the necessity to create a new type of timer ship every time one wants the player to be in a system for a different amount of time, I'd suggest creating some sort of mission objective that requires the player to stay within a system for a certain amount of time, most likely measured in seconds. 

An example of where this could be used in the game currently is "Wanderers: Mereti Observation". Currently this mission's objective is complete if a specific Sestor ship becomes disabled, but I have had instances where the ship is disabled, the game tells me I'm good to go, but in the time that it takes for me to leave the system, the ship gets repaired and suddenly I'm unable to complete the mission because the ship is no longer disabled. (Or at least that's what I believe is occurring, because I've had this mission trigger the dialog multiple times in a single go before.) Having the player stay in the system for say a minute or two before receiving the dialog that they're good to go would be a much cleaner system.

**Describe alternatives you've considered**
An alternative that I've thought of would be to allow negative hull repair values that will damage a ship over time. This would allow timer ships to have a massive amount of shields as to never get destroyed by an explosion as they slowly lose hull until becoming disabled and triggering the disabled event. As mentioned previously though, this has the problem of having to create a new timer ship every time a mission creator wants to use a different amount of time for the player to be in system.

**Comment by Zitchas** [comment]
Yes! The timer ship has always struck me as a creative use of in-game mechanics to accomplish something that the in-game mechanics were never meant to do. And it shows in how unreliable it is. Doesn't work if the player moves around, doesn't work if they hang out too far from the system center. The timer might get short-circuited by an exploding ship....

We really need to simply get a proper timer in game, and this "stay in system for X" would be a great way to do it.

**Comment by DmitrySukhv** [comment]
Good idea. Also, it can be used in brand new mission type like 'Patrol this sector for N minutes for not a single merchant ship gets destroyed by pirates' (taken from Space Rangers 2D game). One drawback: the timeline of ES is somewhat screwed - you use real-time while in system and just turn a day when moving to another system, so entire 'Stay here guarding for 2 minutes' thing looks rather ridiculous.

**Comment by Amazinite** [comment]
Copy pasting over some stuff I mentioned on Discord about this feature.

> `timer <seconds>` should be possible, but the `timer` node should also be able to have children that specify certain parameters for how the timer counts, such as needing to have a low velocity, or being in combat, or being cloaked or not, or how the timer resets, if at all. In this case, we'd want a timer that counts down only when you're relatively stationary, but resets if you leave or start moving around too much.
>
>But there's a mission in the Wanderer campaign where we'd just want a simple timer of "remain in Mesuket for X seconds" and leaving shouldn't reset it.
>Although perhaps for that mission it should be "X seconds while within Y distance from the system center" so that flying off into nothingness doesn't let you finish the mission as you can currently.
>Because you're supposed to be observing the automata.
>A mission should also be able to have multiple timers, with each timer being able to trigger an action upon >completion. That is, timers shouldn't just simply be a mission objective.
I> imagine timers being more like `npc` nodes than `destination` nodes.

**Comment by Zitchas** [comment]
Yeah.... Last time I did "Sad Archie," All my attempts at just waiting it out went nowhere. I ended up lining up the blinking dot with an asteroid, and shooting the asteroid with heavy rockets.
